### DISCUSSION TITLE
Not able to connect with ibm i in vscode getting Not connected to pub400.com! Check your credentials (All configured authentication methods failed) this errror

### DISCUSSION THREAD

**Discussion by swapniltake1** [codefori/vscode-ibmi#2152] [IBM i] [answered]
![image](https://github.com/codefori/vscode-ibmi/assets/61576958/e7fef7bb-ac22-4bfe-aba9-1e1289be7462)
While connecting in vscode am getting this error. answer this what i have to do _Not connected to pub400.com! Check your credentials (All configured authentication methods failed)_

**Comment by sebjulliand** [comment]
Hello @swapniltake1 ,
you seemingly used the wrong user/password; or your password may have expired on PUB400. Try connecting with a 5250 emulator to check that 9and double check your password).

Worst case scenario: your profile may be disabled on PUB400, in which case you'll have to reach out to the admin to sort it out.

**Reply by swapniltake1** [reply]
hi @sebjulliand 
My emulator is functioning correctly, and I have verified that my username and password are accurate. Despite this, I am still unable to connect using IBM i VSCode. Could you please advise on how to resolve this issue?

**Reply by sebjulliand** [reply]
Re-type or copy/paste your password in your connection configuration.
That's the only remaining problem (and the most common for this issue), unless you use a private key to connect.

**Reply by swapniltake1** [reply]
how can we generate private key in pub400

**Comment by thomprl** [comment]
are you using the correct port?  pub400 uses a different port.
https://pub400.com/cgi/faqs.nd/go?yCURCAT=1&yCURFAQ=2

**Reply by swapniltake1** [reply]
yes am using correct port but still same issue.
![image](https://github.com/codefori/vscode-ibmi/assets/61576958/eecb393e-17d1-4e43-a737-02546e870ebf)

**Reply by chrjorgensen** [reply, marked as answer]
Some suggestions:

1. Change your password via 5250 and see if it changes anything.
2. Check your home directory in 5250: `WRKLNK '.'`. It should be `/home/swapnil10`.
3. Check the files in your home directory - that you own all the files. You can change ownership via 5250 command `CHGOWN OBJ('/home/swapnil10/*') NEWOWN(SWAPNIL10) SUBTREE(*ALL)`
4. Run your SSH client in verbose mode: `ssh -vvv swapnil10@pub400.com -p 2222` and check for any suspicious errors.

Code for IBM i is merely a SSH client like many other SSH clients, but you get more error information when using a regular SSH client. Make your login work in the SSH client - when it works here, it will also work in Code for IBM i.

If you can't make it work, please contact PUB400 for support.
